src/hal/src/format.rs

```diff
@@ -148,10 +148,12 @@ bitflags!(
     pub struct ImageFeature: u32 {
         /// Image view can be sampled.
         const SAMPLED = 0x1;
-        /// Image view can be used as storage image.
+        /// Image view can be used as storage image with exclusive read & write access.
         const STORAGE = 0x2;
-        /// Image view can be used as storage image (with atomics).
+        /// Image view can be used as storage image with atomics.
         const STORAGE_ATOMIC = 0x4;
+        /// Image view can be used as storage image with simultaneous read/write access.
+        const STORAGE_READ_WRITE = 0x8;
```

Comment:
what about Metal? can we have some logic to check when this flag is available?

Reply:
Yes, absolutely. We need to check OSX_ReadWriteTextureTier / MTLReadWriteTextureTier and then grind through [this table](https://developer.apple.com/metal/Metal-Feature-Set-Tables.pdf).
I haven't done anything yet though for OpenGL and Metal, since the `format_properties` properties aren't implemented there to begin with, so there needs to be a follow up to this PR eventually to fill them out.

(got this all via one of these amazing writeups the WebGPU specification effort produces https://github.com/gpuweb/gpuweb/issues/513)

Reply:
That's not true. `format_properties` are implemented. Maybe you overlooked it?

Reply:
ouch, ofc you're right. Apologies, I must have looked at the gl version twice.

Turns out @bsecura added MTLReadWriteTextureTier to metal-rs quite recently, perfect timing \o/
Only means that we need to update to a newer version now. So I went and put in the git repo in cargo.toml for testing. Had a bit of ripple effects of things that need to be updated - want me to put that commit into a separate PR?

Ofc we can't land this like this. I saw you did some stuff in metal-rs yourself recently and already bumped the version number to 0.20.1, mind publishing to crates.io timely so this here can move on? :)


Sidenote: I found that there's also such a feature in the gpufamily->features pdf which metal-rs also implements, but in the pdf this comes with footnotes attached one should please check on the texture tier, a bit odd situation, seems the TextureTier enum is simply newer.

Reply:
#3561 already switches to metal-rs master. I'll get it in now.

Reply:
ah didn't see that. I'll rebase then!

Reply:
done.